CMakeLists.txt

```diff
@@ -1,6 +1,9 @@
 cmake_minimum_required(VERSION 3.1)
 project(apriltag VERSION 3.1.0 LANGUAGES C CXX)
 
+if(POLICY CMP0077)
+    cmake_policy(SET CMP0077 NEW)
+endif()
```

Comment:
What is policy `CMP0077` needed for? Does this require a newer CMake version?

Reply:
It does not require a new version of CMake. It was to suppress a warning from CMake when using version 3.13 or greater, as the behavior of `option()` was changed slightly, as noted [here](https://cmake.org/cmake/help/latest/policy/CMP0077.html). Should the minimum CMake version ever be increased to 3.13 or higher this can be removed.

Reply:
But if `option` is removed in this PR, then we don't need this.

Reply:
`option` is not removed in this PR. The behavior of `option` changed in CMake version 3.13. Prior to 3.13 `option` would disregard non-cached variables. 

For example, in 3.12 the following snippet would result in `BUILD_SHARED_LIBS` having a value of `ON`. In 3.13 and later it would have a value of `OFF`.

```cmake
set(BUILD_SHARED_LIBS OFF)
option(BUILD_SHARED_LIBS "Build shared libraries" ON)
```

Setting `CMP0077` indicates to CMake versions 3.13 and greater we explicitly intend to use the new behavior and not to issue a warning about it.

Reply:
I might be wrong here, but [`BUILD_SHARED_LIBS`](https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html) is already a global CMake variable and specifying it again as `option` here is redundant. The condition for `BUILD_PYTHON_WRAPPER` should jus work with the global `BUILD_SHARED_LIBS` without redefining it.

Reply:
`BUILD_SHARED_LIBS` does not have a default value. If you omit `option` it will build statically by default. In order to maintain the default behavior of the library it has to be included. The [documentation](https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html) states, "This variable is often added to projects as an option() so that each user of a project can decide if they want to build the project using shared or static libraries."

Reply:
You will only get a warning if `BUILD_SHARED_LIBS` is already defined as a non-cache variable. Add 'set(BUILD_SHARED_LIBS OFF)` directly above the option and you will see it.

Reply:
@simogasp Although ultimately I suppose the policy can also be set by the parent project that is linking against Apriltag. I'm not sure what the CMake standard approach is, to have each library set its policies or to rely on the parent setting policies?

@christian-rauch You are correct, it will build the static version if not defined (at least it did when I tested it, I don't know if the behavior is officially defined). Regardless of its behavior when not defined, it is typical to define it so there is always a specified default behavior.

Reply:
Ok, in summary:
- we need the `option` to change the default behaviour to "shared"
- we need to set `CMP0077` to define the default `option` behaviour

I am happy with this.
@Robaire Can you maybe extend the commit message to document why setting `CMP0077` is required?

Reply:
@christian-rauch Done

Reply:
Thanks. And sorry for being pedantic about the `CMP0077` :-)